Fluid/Ast/MemberExpression.cs

```diff
@@ -4,28 +4,33 @@ namespace Fluid.Ast
 {
     public sealed class MemberExpression : Expression
     {
-        public MemberExpression(MemberSegment segment)
+        private readonly MemberSegment[] _segments;
+
+        public MemberExpression(MemberSegment segment) : this([segment])
+        {
+        }
+
+        public MemberExpression(IReadOnlyList<MemberSegment> segments) : this(segments as MemberSegment[] ?? segments.ToArray())
         {
-            Segments = [segment];
         }
 
-        public MemberExpression(IReadOnlyList<MemberSegment> segments)
+        internal MemberExpression(MemberSegment[] segments)
         {
-            Segments = segments ?? [];
+            _segments = segments ?? [];
 
-            if (Segments.Count == 0)
+            if (_segments.Length == 0)
             {
-                throw new ArgumentException("At least one segment is required in a MemberExpression");
+                ExceptionHelper.ThrowArgumentNullException(nameof(segments), "At least one segment is required in a MemberExpression");
```

Comment:
```suggestion
                ArgumentNullException.ThrowIfNull(nameof(segments));
```

Reply:
Only available from NET 7 onwards.

Reply:
Right, I forgot that the library targets multiple TFM, but we could use `ArgumentNullException.ThrowIfNull` in the helper you created if the TFM >= .NET 7

Reply:
I don't understand the need for such complexity for a case that should never even be hit.

Reply:
Leave it then